packages/iov-bcp-types/src/transactions.ts

```diff
@@ -44,131 +37,62 @@ export interface Amount {
   readonly tokenTicker: TokenTicker;
 }
 
-export interface ChainAddressPair {
-  readonly chainId: ChainId;
-  readonly address: Address;
-}
-
-export enum TransactionKind {
-  AddAddressToUsername,
-  Send,
-  /** @deprecated see SetNameTx */
-  SetName,
-  SwapOffer,
-  SwapCounter,
-  SwapClaim,
-  SwapTimeout,
-  RegisterBlockchain,
-  RegisterUsername,
-  RemoveAddressFromUsername,
-}
-
-export interface BaseTx {
+/** The basic transaction type all transactions should extend */
+export interface UnsignedTransaction {
+  /**
+   * The domain in which the concrete transaction is valid
+   *
+   * This should be used for type detection only and not be encoded somewhere.
+   * Right now we use "bns", "ethereum", "lisk" and "rise" but this could also
+   * be migrated to a Java-style package names like "io.lisk.mainnet" later on.
+   */
+  readonly domain: string;
   /** the chain on which the transaction should be valid */
   readonly chainId: ChainId;
   readonly fee?: Amount;
   readonly gasPrice?: Amount;
   readonly gasLimit?: Amount;
   // signer needs to be a PublicKey as we use that to as an identifier to the Keyring for lookup
   readonly signer: PublicKeyBundle;
-  readonly ttl?: TtlBytes;
-}
-
-export interface AddAddressToUsernameTx extends BaseTx {
-  readonly kind: TransactionKind.AddAddressToUsername;
-  /** the username to be updated, must exist on chain */
-  readonly username: string;
-  readonly payload: ChainAddressPair;
 }
 
-export interface SendTx extends BaseTx {
-  readonly kind: TransactionKind.Send;
+export interface SendTransaction extends UnsignedTransaction {
+  readonly kind: "send";
```

Comment:
Yeah, I learned with redux actions that these enum descriptions are just verbose and don't add type safety. Cleaner this way